### PR TITLE
Migrate intellij tests and context_test to null safety

### PR DESCRIPTION
**Pull request by jmagman**
Part of #71511

### CODE REVIEW

**Comment by jmagman** [comment]
These aren't really related files, am trying to keep these migration reviews relatively small.

File: packages/flutter_tools/test/general.shard/intellij/intellij_validator_test.dart [hunk 1]
```diff
@@ -379,7 +377,7 @@ void createIntellijFlutterPluginJar(String pluginJarPath, FileSystem fileSystem,
   flutterPlugins.addFile(ArchiveFile('META-INF/plugin.xml', flutterPluginBytes.length, flutterPluginBytes));
   fileSystem.file(pluginJarPath)
     ..createSync(recursive: true)
-    ..writeAsBytesSync(ZipEncoder().encode(flutterPlugins));
+    ..writeAsBytesSync(ZipEncoder().encode(flutterPlugins)!);
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/intellij/intellij_validator_test.dart:380]: ZipEncoder returns `null`?

Comment (jmagman) [reply packages/flutter_tools/test/general.shard/intellij/intellij_validator_test.dart:380]: `encode` can https://pub.dev/documentation/archive/latest/archive_io/ZipEncoder/encode.html

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/intellij/intellij_validator_test.dart:380]: what a yucky api :(

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**